fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/provider/TranslationConsumer.java

```diff
@@ -0,0 +1,146 @@
+	 * @param block The {@link Block} to get the language entry key from.
+	 * @param value The value of the entry.
+	 */
+	default void add(Block block, String value) {
+		add(block.getTranslationKey(), value);
+	}
+
+	/**
+	 * Adds a language entry for an {@link ItemGroup}.
+	 * @param group The {@link ItemGroup} to get the language entry key from.
+	 * @param value The value of the entry.
+	 */
+	default void add(ItemGroup group, String value) {
+		add("itemGroup." + group.getName(), value);
+	}
+
+	/**
+	 * Adds a language entry for an {@link EntityType}.
+	 * @param entityType The {@link EntityType} to get the language entry key from.
+	 * @param value The value of the entry.
+	 */
+	default void add(EntityType<?> entityType, String value) {
+		add(entityType.getTranslationKey(), value);
+	}
+
+	/**
+	 * Adds a language entry for an {@link Enchantment}.
+	 * @param enchantment The {@link Enchantment} to get the language entry key from.
+	 * @param value The value of the entry.
+	 */
+	default void add(Enchantment enchantment, String value) {
+		add(enchantment.getTranslationKey(), value);
+	}
+
+	/**
+	 * Adds a language entry for an {@link EntityAttribute}.
+	 * @param entityAttribute The {@link EntityAttribute} to get the language entry key from.
+	 * @param value The value of the entry.
+	 */
+	default void add(EntityAttribute entityAttribute, String value) {
+		add(entityAttribute.getTranslationKey(), value);
+	}
+
+	/**
+	 * Adds a language entry for a {@link StatType}.
+	 * @param statType The {@link StatType} to get the language entry key from.
+	 * @param value The value of the entry.
+	 */
+	default void add(StatType<?> statType, String value) {
+		add(statType.getTranslationKey(), value);
+	}
+
+	/**
+	 * Adds a language entry for a {@link StatusEffect}.
+	 * @param statusEffect The {@link StatusEffect} to get the language entry key from.
+	 * @param value The value of the entry.
+	 */
+	default void add(StatusEffect statusEffect, String value) {
+		add(statusEffect.getTranslationKey(), value);
+	}
+
+	/**
+	 * Adds a language entry for an {@link Identifier}.
+	 * @param identifier The {@link Identifier} to get the language entry key from.
+	 * @param value The value of the entry.
+	 */
+	default void add(Identifier identifier, String value) {
+		add(identifier.toTranslationKey(), value);
+	}
+
+	/**
+	 * Merges an existing language file into the generated language file.
+	 * @param existingLanguageFile The path to the existing language file.
+	 * @throws IOException If loading the language file failed.
+	 */
+	default void add(Path existingLanguageFile) throws IOException {
+		Gson gson = new Gson();
+
+		JsonObject langEntryJson = gson.fromJson(Files.readString(existingLanguageFile), JsonObject.class);
+
+		for (Map.Entry<String, JsonElement> stringJsonElementEntry : langEntryJson.entrySet()) {
+			add(stringJsonElementEntry.getKey(), stringJsonElementEntry.getValue().getAsString());
+		}
+	}
+}
```

Comment:
Catching `IOException` sounds like an unnecessary chore IMO, what would you do if it fails anyway? Most will probably end up rethrowing `RuntimeException`.

Also nitpicky, but this method can be made more compact.
```java
default void add(Path existingLanguageFile) {
	try (Reader reader = Files.newBufferedReader(existingLanguageFile)) {
		JsonObject translations = JsonParser.parseReader(reader).getAsJsonObject();

		for (String key : translations.keySet()) {
			add(key, translations.get(key).getAsString());
		}
	} catch (IOException e) {
		throw new RuntimeException(e);
	}
}
```